### PR TITLE
Update King to 2.2.7

### PR DESCRIPTION
**Pull request by mtekman**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Based on PR https://github.com/galaxyproject/tools-iuc/pull/4493

It adds extra parameters, sections, and makes many UI improvements

### CODE REVIEW

**Review by bgruening — COMMENTED** [review]
Can you maybe add expected_num_outputs and maybe add a test for the new collection outputs?

**Comment by bernt-matthias** [comment]
Supersedes https://github.com/galaxyproject/tools-iuc/pull/4493 

@mtekman you should also be able to add to the PR opened by the planemo-autoupdate bot. Tell me if I should tell you how...

**Comment by mtekman** [comment]
@bernt-matthias yes please -- I tried pushing to it, but it rejected it

**Comment by bernt-matthias** [comment]
@mtekman .. I forgot that only IUC members can push to branches of the auto update bot. So please continue over here.

File: tools/king/king.xml [hunk 1]
```diff
@@ -2,82 +2,266 @@
...
-    $related
-    $duplicate
-    $kinship
-    $ibdseg
-    $ibs
-    $homog
-    #if str($degree):
-    --degree $degree
+    #if $opt_int.family:
+    ln -s '$opt_int.family' input.fam &&
```
Comment (bernt-matthias) [inline tools/king/king.xml:21]: Can you add indentation for `if`?

File: tools/king/king.xml [hunk 2]
```diff
@@ -2,82 +2,266 @@
...
+    --sexchr $opt_int.sexchr
+    #end if
+    ## Output
+    $opt_par.rplot
+    $opt_par.pngplot
+    $opt_par.plink
     > '$kingoutlog'
     ]]></command>
     <inputs>
         <param name="genotype" type="data" format="pbed,binary" label="Binary Genotype File" />
```
Comment (bernt-matthias) [inline tools/king/king.xml:111]: Is `binary` here useful? This allows nearly all file types?

**Review by bgruening — APPROVED** [review]

**Comment by bernt-matthias** [comment]
Superb @mtekman

**Pull request merged**